modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -230,47 +229,41 @@ object GmosControllerEpics extends GmosEncoders {
   def setWavelength(state: GmosCCEpicsState, w: Length): Option[IO[Unit]] =
     applyParam(state.disperserWavel, encode(w), CC.setDisperserLambda)
 
-  def setDisperserParams[T <: GmosController.SiteDependentTypes: Encoders](state: GmosCCEpicsState, cfg: GmosController.Config[T], d: GmosController.Config[T]#GmosDisperser): List[Option[IO[Unit]]] = {
+  def setDisperserParams[T <: GmosController.SiteDependentTypes: Encoders](state: GmosCCEpicsState,
+                                                                           cfg: GmosController.Config[T],
+                                                                           d: GmosController.Config[T]#GmosDisperser)
+  : List[Option[IO[Unit]]] = {
     val set: List[Option[IO[Unit]]] = d match {
-      case cfg.GmosDisperser.Mirror =>
-        val s: Option[IO[Unit]] = setDisperser(state, cfg.mirror)
-        //If disperser is set, force mode configuration
-        s.foldMap(_ => List(s, CC.setDisperserMode(disperserMode0).some))
+      case cfg.GmosDisperser.Mirror => List(setDisperser(state, cfg.mirror))
       case cfg.GmosDisperser.Order0(d) =>
         val s0: Option[IO[Unit]] = setDisperser(state, d)
         // If disperser is set, force order configuration
-        val s: List[Option[IO[Unit]]] = if(s0.isEmpty) List(setOrder(state, Order.ZERO))
-                else CC.setDisperserOrder(disperserOrderEncoder.encode(Order.ZERO)).some :: List(s0)
-        //If disperser or order are set, force mode configuration
-        s.foldMap(_ => s ++ List(CC.setDisperserMode(disperserMode0).some))
+        if(s0.isEmpty) List(setOrder(state, Order.ZERO))
+        else CC.setDisperserOrder(disperserOrderEncoder.encode(Order.ZERO)).some :: List(s0)
       case cfg.GmosDisperser.OrderN(d, o, w) =>
         val s0: Option[IO[Unit]] = setDisperser(state, d) // None means disperser is already in the right position
-        val s: List[Option[IO[Unit]]] =
-          // If disperser is set, the order has to be set regardless of current value.
-          if (s0.isEmpty) {
-            List(
-              setOrder(state, o),
-              setWavelength(state, w)
-            )
-          } else {
-            List(
-              CC.setDisperserOrder(disperserOrderEncoder.encode(o)).some,
-              s0,
-              setWavelength(state, w)
-            )
-          }
-
-        //If disperser, order or wavelength are set, force mode configuration
-        s.foldMap(_ => s ++ List(CC.setDisperserMode(disperserMode0).some))
```

Comment:
`s` here was a non empty `List`, not an `Option` ! This caused the Mode to be set always.